libecho/src/Packet.cc

```diff
@@ -0,0 +1,117 @@
+#include "Packet.h"
+
+// #define crcpp_uint32
+#include <arpa/inet.h>
+#include "CRC.h"
+
+Packet Packet::loadHeaderFromBytes(const std::vector<uint8_t> &bytes) {
+    Packet p;
+
+    if (bytes.size() != HEADER_SIZE) {
+        throw IncorrectFormat();
+    }
+
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-cstyle-cast)
+    p.flags = ntohs(((uint16_t *)bytes.data())[0]);
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-cstyle-cast)
+    p.size = ntohs(((uint16_t *)bytes.data())[1]);
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-cstyle-cast)
+    p.number = ntohs(((uint16_t *)bytes.data())[2]);
```

Comment:
Maybe those casts should be moved to function, se we only need to exclude one line from linting?

Reply:
Resolved offline ;)